### PR TITLE
Add mutation-based fuzzer

### PR DESCRIPTION
**Pull request by mingxguo27**
Due to a Github error, I'm creating this PR in substitution for the old PR #68. Sorry for the reviewing trouble!
PR contains a MVP for the mutation-based fuzzer and unit tests for the mutator.

### CODE REVIEW

**Comment by kaczmarczyck** [comment]
Close for now as long as #68 keeps working.

**Comment by kaczmarczyck** [comment]
#68 stopped working again, we will need to re-upload the code here.

File: src/fuzzing/mutator_test.cc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  std::vector<uint8_t> data = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+  for (int i = 0; i < 100; ++i) {
+    srand(i);
+    size_t expected_current_size = data.size();
+    std::multiset<uint8_t> data_before_mutation(data.begin(), data.end());
+    ASSERT_FALSE(mutator::ShuffleBytes(data, /* max_size = */ 1));
+    ASSERT_TRUE(mutator::ShuffleBytes(data, expected_current_size));
+    ASSERT_TRUE(data.size() == expected_current_size);
+    std::multiset<uint8_t> data_after_mutation(data.begin(), data.end());
+    ASSERT_TRUE(data_before_mutation == data_after_mutation);
```
Comment (jennypullman) [inline src/fuzzing/mutator_test.cc:75]: Shouldn't this be false?

**Review by jennypullman — CHANGES_REQUESTED** [review]

File: src/fuzzing/mutator_test.cc (hunk 1 above)
Comment (mingxguo27) [reply src/fuzzing/mutator_test.cc]: I'm comparing the set here so shuffle should keep the elements set invariant. I added the change check in the new commit

**Review by kaczmarczyck — APPROVED** [review]

File: src/fuzzing/mutator_test.cc (hunk 1 above)
Comment (jennypullman) [reply src/fuzzing/mutator_test.cc]: Oop thanks!

**Review by jennypullman — APPROVED** [review]

**Pull request merged**